src/diff.jl

```diff
@@ -465,12 +465,27 @@ function sparsejacobian(ops::AbstractVector, vars::AbstractVector; simplify=fals
     sp = jacobian_sparsity(ops, vars)
     I,J,_ = findnz(sp)
 
+    exprs = sparsejacobian_vals(ops, vars, I, J, simplify=simplify)
```

Comment:
If this is for values, then why is this here? The only thing calculated by the `sparsejacobian` call is the structure

Reply:
Using the result of `sparsejacobian_vals` you can get the values.
The idea is: first, you compute the rows and cols, so you know the number of non-zeros ; then, if required by the user, you call `sparsejacobian_vals` and then get the values. So, I avoid having the SparseMatrix structure.

Reply:
Oh I get it now. Why avoid the sparse matrix structure though? That seems more intuitive?

Reply:
Oh, because Julia sparse only supports Floats. I see, yeah this makes sense.

Reply:
So, this changes wouldn't change the expected workflow, and gives an additional possibility.